### ISSUE TITLE
Ruleset listing endpoint: Add paths parameter

### ISSUE THREAD

**Issue by asdine** [heetch/regula#22]
The `GET /ruletsets/<prefix>?list&paths` endpoint must return only the ruleset paths and not the ruleset themselves.
The `paths` option is a boolean.

The payload must look like this:

```json
{
  "revision": "2087",
  "rulesets": [
    {
      "path": "a/b/c",
    },
    {
       "path": "d/e/f",
    }
  ]
}
```

The payload is the same as without this option except that only the `path` variable is returned.

**Comment by asdine** [comment]
Edited, thank you :+1:

**Comment by YassineE** [comment]
Do you validate this:
- `GET /ruletsets/<prefix>?list&paths`: returns paths only
- `GET /ruletsets/<prefix>?list&paths=true`: returns paths only
- `GET /ruletsets/<prefix>?list&paths=false`: current behaviour

**Comment by asdine** [comment]
Not really, we must do the same as we already do with the other endpoints: https://github.com/heetch/regula/blob/master/api/server/api.go#L26-L59

**Comment by YassineE** [comment]
Allright, so we don't care if it's a boolean or not. 
> The paths option is a boolean.

This makes me think that.

**Comment by asdine** [comment]
I understand, basically the presence of the param is what enables endpoint, regardless of the content. This is what Consul is doing https://www.consul.io/api/kv.html#read-key

**Comment by YassineE** [comment]
> This is specified as part of the URL as a query parameter.

TIL 👍

**Comment by YassineE** [comment]
Should we also handle the `limit` parameter next to the `paths` one?
For example, is `GET /ruletsets/<prefix>?list&paths&limit=10` a valid endpoint?

**Comment by asdine** [comment]
Yes, pagination must also work

**Comment by YassineE** [comment]
@asdine I am facing an issue with the paging. We can’t rely on the `WithLimit` option of etcd because if we will retrieve different paths for each version.
For example, if we have a `limit == 50` and the first 30 keys on etcd correspond to the same rulesets but with different versions, at the end, we will have 21 different paths which are different from what the client asks.
I tried to handle the `limit` by hand but it’s the same problem as above. I think the last option is to query several times etcd until we have the correct numbers of paths requested by the caller. But it ends with a few calls to etcd. WDYT?

**Comment by asdine** [comment]
In that case, we can skip the pagination. Also, we already have the `rulesets/latest` prefix that can be used to list all the paths, that way we have a light way of getting all the existing rulesets paths without even loading the value of the each of the paths (cf KeysOnly option, or something like that)

**Comment by YassineE** [comment]
> we already have the rulesets/latest prefix that can be used to list all the paths

The problem with `rulesets/latest` is that it requires an absolute path whereas for this endpoint we are dealing with prefixes. So the `rulesets/latest` is useless here IMO.

> cf KeysOnly option

It is what I am currently using.

> In that case, we can skip the pagination.

Allright, let's skip the pagination for this.

**Comment by asdine** [comment]
> The problem with rulesets/latest is that it requires an absolute path whereas for this endpoint we are dealing with prefixes. So the rulesets/latest is useless here IMO.

Are you talking about the etcd prefix `rulesets/latest`?
In etcd, we have:

|key|value|
|-|-|
|`rulesets/<path>/<version>`|ruleset entries|
|`rulesets/latest/<path>`|version of the latest ruleset|

**Comment by YassineE** [comment]
> Are you talking about the etcd prefix rulesets/latest?

No, i was talking about the `Latest` method of the store but in the end it's the same, the `rulesets/latest` deals with a path not with a prefix.
Indeed, under the `rulesets/latest` you will have the path of the latest version of **all** rulesets thus you can have all the paths. But we can't use it to retrieve all the paths under a given prefix.
For example, let's say in etcd I have 3 rulesets (sorted in asc order):
- `boost/canary-go/v1`
- `boost/canary-go/v2`
- `boost/canary-ex/v1`

In the `rulesets/latest` under the prefix `boost` I will have only `boost/canary-ex/v1`.
Am I clear?

**Comment by asdine** [comment]
No you won't, you'll have:
- `boost/canary-go`
- `boost/canary-ex`

**Comment by YassineE** [comment]
Indeed, anyway we skip the pagination. Even with the `rulesets/latest`, it is complicated to deal with.

**Comment by YassineE** [comment]
@asdine finally I find a way to deal with the pagination. So we will be able to paginate the `paths` endpoint. 👍

**Comment by YassineE** [comment]
Fixed in #43.
